@vstinner, I tried running tests but sadly the build failed due to what seems like unrelated issue:
```
/builds/cpython-main/Modules/termios.c: In function 'termios_tcgetwinsize_impl':
/builds/cpython-main/Modules/termios.c:389:9: error: implicit declaration of function 'ioctl' [-Werror=implicit-function-declaration]
389 |     r = ioctl(fd, TIOCGWINSZ, &w);
    |         ^~~~~
/builds/cpython-main/Modules/_multiprocessing/multiprocessing.c: In function 'multiprocessing_exec':
/builds/cpython-main/Modules/_multiprocessing/multiprocessing.h:44:32: error: implicit declaration of function 'sysconf'; did you mean '_sysconf'? [-Werror=implicit-function-declaration]
44 |         # define SEM_VALUE_MAX sysconf(_SC_SEM_VALUE_MAX)
   |                                ^~~~~
```
but I don't see how this change can break anything as it's how we compile Python for many years now.

Just two minor nits for the NEWS fragment - it's at least since 2005 (when Solaris was opened) and it's not just Oracle Solaris but Solaris in general.

--- a/Misc/NEWS.d/next/Build/2023-09-02-18-04-15.gh-issue-63760.r8hJ6q.rst
+++ b/Misc/NEWS.d/next/Build/2023-09-02-18-04-15.gh-issue-63760.r8hJ6q.rst
@@ -1,0 +1,3 @@
+Fix Solaris build: no longer redefine the ``gethostname()`` function. Solaris
+defines the function since 2005. Patch by Victor Stinner, original patch by
+Jakub Kulík.

--- a/Modules/socketmodule.c
+++ b/Modules/socketmodule.c
@@ -111,11 +111,6 @@ Local naming conventions:
 #include "pycore_fileutils.h"     // _Py_set_inheritable()
 #include "pycore_moduleobject.h"  // _PyModule_GetState
 
-// gethostname() prototype missing from Solaris standard header files
-#ifdef __sun
-extern int gethostname(char *, int);
-#endif
-
 #ifdef _Py_MEMORY_SANITIZER
 #  include <sanitizer/msan_interface.h>
 #endif

